src/jit/lsra.cpp

```diff
@@ -1809,11 +1809,16 @@ void LinearScan::doLinearScan()
 
 void LinearScan::recordVarLocationsAtStartOfBB(BasicBlock* bb)
 {
+    if (!enregisterLocalVars())
+    {
+        return;
+    }
     JITDUMP("Recording Var Locations at start of BB%02u\n", bb->bbNum);
     VarToRegMap map   = getInVarToRegMap(bb->bbNum);
     unsigned    count = 0;
 
-    VARSET_ITER_INIT(compiler, iter, bb->bbLiveIn, varIndex);
+    VarSetOps::AssignNoCopy(compiler, currentLiveVars, VarSetOps::Intersection(compiler, registerCandidateVars, bb->bbLiveIn));
```

Comment:
I assume you mean the ration of reg-candidate lclVars to tracked lclVars (the difference between tracked and untracked is illustrated by the improvements Bruce got with #11698. I think there are a non-trivial number of tracked lclVars that are not reg candidates (e.g. non-enregisterable structs that are not promoted). I think that your suggestion could be considered as part of #11339.